File: go/pkg/two_sum.go
package pkg

func twoSum(numbers []int, target int) []int {
	seen_num := make(map[int]int)
	for i, num := range numbers {
		seen_num[num] = i
	}

	for _, num := range seen_num {
		_, ok := seen_num[target-num]
		if ok {
			return []int{seen_num[num] + 1, seen_num[target-num] + 1}
		}
	}

	return []int{-1, -1}
}

File: python/two-sum.py
from typing import List

def twoSum(numbers: List[int], target: int) -> List[int]:
    seen_nums = {}
    for index, num in enumerate(numbers):
        seen_nums[num] = index

    for num in seen_nums:
        if(target-num in seen_nums):
            return [seen_nums[num]+1, seen_nums[target-num]+1]
    return [-1,-1]

if __name__ == "__main__":
    print(twoSum([2,7,11,15], 9))